karyon3-admin-simple/src/main/java/com/netflix/karyon/admin/ui/AdminUIHttpHandler.java

```diff
@@ -25,58 +20,38 @@
 public class AdminUIHttpHandler implements HttpHandler {
     private static final Logger LOG = LoggerFactory.getLogger(AdminUIHttpHandler.class);
 
-    private final ConcurrentMap<String, String> templates = new ConcurrentHashMap<>();
     private final AdminUIServerConfig config;
-    private final MimetypesFileTypeMap mimeTypes;
+    private final StaticResourceProvider provider;
     
     @Inject
     public AdminUIHttpHandler(AdminUIServerConfig config) {
         this.config = config;
-        InputStream is = null;
-        try {
-            is = this.getClass().getResourceAsStream(config.mimeTypesResourceName());
-            if (is != null) {
-                mimeTypes  = new MimetypesFileTypeMap(is);
-            }
-            else {
-                throw new RuntimeException("Unable to load mime.types file");
-            }
-        }
-        finally { 
-            if (is != null) {
-                try {
-                    is.close();
-                } catch (IOException e) {
-                    LOG.debug("Unable to close mime type file");
-                }
-            }
-        }
+        this.provider = 
+            new CachingStaticResourceProvider(
+                new FileSystemResourceProvider(
+                    config.resourcePath(),
+                    config.mimeTypesResourceName()));
     }
     
     @Override
     public void handle(HttpExchange arg0) throws IOException {
         String path = arg0.getRequestURI().getPath();
-        LOG.info("'{}'", path);
+        LOG.debug("'{}'", path);
         
+        StaticResource resource;
         try {
-            String resource = String.format("/%s%s", config.resourcePath(), path);
-            String content = getResource(resource);
-            if (content != null) {
-                String mimeType  = mimeTypes.getContentType(resource);
-                writeResponse(arg0, 200, content, mimeType);
+            resource = provider.getResource(path).get();
+            if (resource.isValid()) {
```

Comment:
Using Optional well would be more likely to have the check then use case respected.